### PR TITLE
Group 1 - Refactoring Library class

### PR DESCRIPTION
**Pull request by PtrBld**

### CODE REVIEW

File: library.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
+                    if max is not None and max.duration < b.duration:
+                        max = b
+                    # end if
+                # end if
+            else:
```
Comment (paolo-notaro) [inline library.py:25]: empty else block not necessary

File: library.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
+                    if max is not None and max.duration < b.duration:
+                        max = b
+                    # end if
```
Comment (paolo-notaro) [inline library.py:23]: end if comments unnecessary

File: library.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            # end else
+        # end for
+
+        print(f"Title: {max.title}")
+        print(f"Author: {max.authors}")
+        print(f"Duration: {max.duration}")
+        print(f"______________________________")
+        return str(max)
+
+    # Constructor
```
Comment (syncall) [inline library.py:36]: All comments in this file are obvious and can be removed

File: library.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    def __init__(self, books: list[Book]):
+        self._books = books
+
+    _books: list[Book] = []
```
Comment (paolo-notaro) [inline library.py:40]: unused constant (_books != self._books)

File: library.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
+    def print_longest_crime_audio_book(self) -> str:
+        crime_audio_books: list[Book] = []
+        max: Optional[Book] = None
```
Comment (syncall) [inline library.py:11]: Bad python style to override python build-in functions. It would be better to rename it

File: library.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+        # print banner
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
```
Comment (paolo-notaro) [inline library.py:20]: Should not use _ variables (non-public variables, see PEP 8)

File: library.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
```
Comment (paolo-notaro) [inline library.py:8]: redundant comment

**Pull request closed without merging**